R/fit_distributions.R

```diff
@@ -1,70 +1,114 @@
-fit_uniform <- function(x, metric=c('mle', 'jaccard', 'intersection', 'ks', 'mse', 'chisq')) {
-  UseMethod('fit_uniform')
-}
 
-#' Fit a uniform distribution to a histogram
+#' Fit the model parameters by optimizing a histogram metric
+#'
+#' @param x numeric vector, representing data to be fit
+#' @param metric a subset of `mle`, `jaccard`, `intersection`, `ks`, `mse`, `chisq`
+#' indicating metrics to use for fit optimization
+#' @param truncated logical, whether to fit truncated distributions
+#' @param distributions character vector indicating distributions,
+#' subset of `norm`, `gamma`, `gamma_flip` and `unif`.
 #'
-#' @param x numeric vector representing the density of a histogram
-#' @param metric one of `jaccard`, `intersection`, `ks`, `mse`, `chisq`
+#' @export
 #'
-#' @return a list with the following data
+#' @return a nested list where each sublist represents a model with the following data
 #' \describe{
 #'     \item{par}{a character string denoting the region_id of the Histogram}
 #'     \item{dist}{the distribution name}
 #'     \item{metric}{the metric used to fit the distribution}
 #'     \item{value}{the fitted value of the metric function}
 #'     \item{dens}{a function that returns the density of the fitted distribution}
 #' }
-#' @exportS3Method fit_uniform numeric
-fit_uniform.numeric <- function(x, metric=c('mle', 'jaccard', 'intersection', 'ks', 'mse', 'chisq')) {
-  # Error checking
-  metric <- match.arg(metric)
+#'
+#' @importFrom DEoptim DEoptim
+fit_distributions <- function(
+    x,
+    metric = c("jaccard", "intersection", "ks", "mse", "chisq"),
+    truncated = FALSE,
+    distributions = c("norm", "gamma", "gamma_flip", "unif")
+) {
+  UseMethod("fit_distributions")
+}
+
+#' @exportS3Method fit_distributions numeric
+fit_distributions.numeric <- function(
+    x,
+    metric = c("jaccard", "intersection", "ks", "mse", "chisq"),
+    truncated = FALSE,
+    distributions = c("norm", "gamma", "gamma_flip", "unif")
+) {
```

Comment:
Did we lose MLE?

Reply:
I added as the default for `fit_distributions` but not for `segment_and_fit` - I think because we were discussing how it conflicts with `max_uniform` 

I can add it as a default for `segment_and_fit`, too but I think there is a bug in this branch that will trigger errors in the tests (one test will trigger a fit for gammas wand it's not properly refactored in this branch).

So tldr: let me know if you want me to add `mle` as a default method in `segment_and_fit` and I will add it to another PR.